src/spec.rs

```diff
@@ -674,25 +524,234 @@ impl Zip64CentralDirectoryEnd {
             number_of_files,
             central_directory_size,
             central_directory_offset,
+            extensible_data_sector,
         } = self;
-        Zip64CDEBlock {
-            magic: Zip64CDEBlock::MAGIC,
-            /* currently unused */
-            record_size: 44,
-            version_made_by,
-            version_needed_to_extract,
-            disk_number,
-            disk_with_central_directory,
-            number_of_files_on_this_disk,
-            number_of_files,
-            central_directory_size,
-            central_directory_offset,
-        }
+
+        (
+            Zip64CDEBlock {
+                magic: Zip64CDEBlock::MAGIC,
+                record_size,
+                version_made_by,
+                version_needed_to_extract,
+                disk_number,
+                disk_with_central_directory,
+                number_of_files_on_this_disk,
+                number_of_files,
+                central_directory_size,
+                central_directory_offset,
+            },
+            extensible_data_sector,
+        )
     }
 
     pub fn write<T: Write>(self, writer: &mut T) -> ZipResult<()> {
-        self.block().write(writer)
+        let (block, comment) = self.block_and_comment();
+        block.write(writer)?;
+        writer.write_all(&comment)?;
+        Ok(())
+    }
+}
+
+pub(crate) struct CentralDirectoryEndInfo {
+    pub eocd: (Zip32CentralDirectoryEnd, u64),
+    pub eocd64: Option<(Zip64CentralDirectoryEnd, u64)>,
+
+    pub archive_offset: u64,
+}
+
+/// Finds the EOCD and possibly the EOCD64 block and determines the archive offset.
+///
+/// In the best case scenario (no prepended junk), this function will not backtrack
+/// in the reader.
+pub fn find_central_directory<R: Read + Seek>(
+    reader: &mut R,
+    archive_offset: ArchiveOffset,
+) -> ZipResult<CentralDirectoryEndInfo> {
+    const EOCD_SIG_BYTES: [u8; mem::size_of::<Magic>()] =
```

Comment:
Do these constants still need to exist in `u32` form? I don't think so.

Reply:
Yeah, having it represented as a u32 makes it so that you have to manually call conversion functions which is entirely unnecessary, because the magic must always be little-endian. I will submit another PR for this, because it is a bit out of scope.